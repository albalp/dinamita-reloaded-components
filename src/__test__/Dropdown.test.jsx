import '@testing-library/jest-dom/extend-expect';
import {render, screen, waitFor, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {prettyDOM} from '@testing-library/dom';

import Dropdown from '../components/dropdown/DropDown';
import { MdArrowDropDown } from 'react-icons/md'

const options = [
    {
      id:1, 
      name: 'Mango',
    },
    {
        id:2, 
        name: 'Fresa',
      }
 ]

describe('Dropdown render', () => {
    it('should render the dropdown', () => {
        expect(render(<Dropdown />)).toMatchSnapshot();
    });

    it('should render the dropdown with title and options and icon', () => {
        render(<Dropdown title="Frutas" options={options} icon={<MdArrowDropDown />}/>)
        const title = screen.getByText(/Frutas/i);
        const inputValue = screen.getByDisplayValue(/Mango/i)

        expect(title).toBeInTheDocument();
        expect(inputValue).toBeInTheDocument();
        
    });
    // it('Should called the click', async()=>{
    //     const handleClick = jest.fn();
    //     render(<Dropdown title="Frutas" options={options} onClick={handleClick} />)
    //     const input = screen.getByDisplayValue(/Mango/i)
    //     expect(input).toBeInTheDocument();
        
    //     userEvent.click(input);
    //     const option = await waitFor(screen.getByText(/Fresa/i));

    //     //expect(input).toHaveBeenCalledWith(option);
    //     userEvent.click(option);
    //     expect(await screen.findByDisplayValue(/Fresa/i)).toBeInTheDocument();
    // });
    
})

describe('Dropdown clicks should work', () => {
    it("The dropdown should be clicked and need show options", async() => {

        render( <Dropdown title="Frutas" options={options} />);
        const dropdown = screen.getByRole('menu', {name: 'dropdown-project'});
        const input = screen.getByPlaceholderText(/Mango/i);
        userEvent.click(dropdown);
        fireEvent.change(input, {target: {value: 'Fresa'}});
        const opt = screen.queryByRole('group', {name: 'group-options-project'});
    
        expect(input).toHaveValue('Fresa');
        expect(opt).toBeInTheDocument();
        
    });

    test('Should to show the options of when do click on the dropdown projects', () => {
        render( <Dropdown title="Frutas" options={options} />);

        const dropdown = screen.getByRole('menu', {name: 'Frutas'});
        fireEvent.click(dropdown);
        const opt = screen.queryByRole('group', {name: 'group-options-project'});
        const option = screen.getByText(/Delivery/i);

        expect(opt).toBeInTheDocument();
        expect(option).toBeInTheDocument();

    });

})