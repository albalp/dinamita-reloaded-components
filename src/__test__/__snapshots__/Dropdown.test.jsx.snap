// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render the dropdown 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dropdown-container false false"
      >
        <p />
        <div
          class="dropdown-size--undefined dropdown  false"
        >
          <div
            aria-label="dropdown-"
            class="select"
            role="menu"
          >
            <input
              class="input-option"
              placeholder=""
              type="text"
              value=""
            />
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m11.998 17 7-8h-14z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dropdown-container false false"
    >
      <p />
      <div
        class="dropdown-size--undefined dropdown  false"
      >
        <div
          aria-label="dropdown-"
          class="select"
          role="menu"
        >
          <input
            class="input-option"
            placeholder=""
            type="text"
            value=""
          />
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m11.998 17 7-8h-14z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
